### DISCUSSION TITLE
Mobile development

### DISCUSSION THREAD

**Discussion by Barshan-Mandal** [picoe/Eto#2130] [Q&A]
how to start development for mobile platforms too using eto.forms

**Comment by cwensley** [comment]
Hey @Barshan-Mandal, unfortunately the iOS and Android platforms have not been kept up to date so aren't very usable right now.  It would take a some effort to get those mobile platforms working again.

The main reason the focus shifted primarily to desktop is that there wasn't enough interest in mobile use with Eto.Forms.  This could change, however.

**Comment by allsorts46** [comment]
Hello! @cwensley, not sure if you remember [this](https://github.com/picoe/Eto/issues/1100) discussion from almost 4 years ago, but I just saw this thread pop up and felt I should probably say something.

Due to my workload I unfortunately never actually followed up on that discussion, but we _did_ in fact end up using Eto as the basis for the Android port of our EPOS software, and I have been maintaining and improving it, privately, for all this time, and it works well.

The reason I have never got around to sharing any of this back to the project is mainly due to the way work has progressed. The priority for any work I've done on this has always been satisfying some specific need for our own project, fixing something that needs to work, adding something we need that's missing. The commits are a mess, all made on one single branch, and when I add or change something new I only ever care if it works on Winforms and Android, not any of the other supported platforms because we don't use them. Also a lot of the UIs I had to port were based on custom-drawn Winforms controls, so much more effort has gone into improving say, Drawable and Graphics, than some of the more common controls, but a good few of those have also been implemented or re-implemented. I did _not_ put any effort into using the Navigation control, just working on the assumption that there is at any time one single top-level Form which comprises the main UI, and if you want to show another 'screen' you create and show another Form or Dialog on top.

I occasionally merge your latest dev branch into ours, to keep it up to date. The last time was only a couple of weeks ago, so it's pretty recent. The biggest issue with merging the other way would be the many miscellaneous changes I've made which I only ever tested on Winforms and Android, might cause problems on the other platforms.

The point is, I'm happy to contribute back anything I've done here that you might consider 'of value', just not sure how much extra work might be created in the process.

@Barshan-Mandal this may or may not be useful for your specific needs, but it is, at least, basic support for Android which does compile and work against the latest version of Eto, and _is_ being used and supported in a real current project.

**Reply by cwensley** [reply]
Hey @allsorts46, thanks for the update!  It's awesome to hear you've been using the Android platform with some success.

Yeah it'd be hard to take a huge merge like that, though I'd be interested to know what changes you have made and maybe some of them could be cherry picked and implemented for the other platforms..   Also, possibly some of the specific stuff you have could be moved out of your Eto branch and into your project so you don't have to deal with merging the code all the time.

That being said, for anything under `Eto.Android/` I don't see a problem with a big merge since it is non-functional anyway.  I guess the first step would be to create a PR and we can figure things out from there, if you're willing.
